frontend/src/views/funding/FundingOverviewView.vue

```diff
@@ -54,47 +176,236 @@
+      [FUNDING_AGREEMENT_STATUS_CODES.ACTIVE]: 'Active',
+      [FUNDING_AGREEMENT_STATUS_CODES.EXPIRED]: 'Expired',
+      [FUNDING_AGREEMENT_STATUS_CODES.TERMINATED]: 'Terminated',
+    }
+    this.resetFilter()
+    await this.loadOrganizationUsers()
+    await this.loadFundingAgreements()
+  },
+  methods: {
+    isValidEndDate,
+
+    resetFilter() {
+      this.selectedFacility = null
+      this.selectedDateFilterType = this.DATE_FILTER_TYPE_VALUES.THREE_MONTHS
+      this.selectedPaymentFilterTypes = [this.PAYMENT_FILTER_TYPES.BASE_FUNDING, this.PAYMENT_FILTER_TYPES.SUPPLEMENTARY_ALLOWANCES, this.PAYMENT_FILTER_TYPES.OTHER]
+      this.startDateFrom = null
+      this.startDateTo = null
+    },
+
+    async loadOrganizationUsers() {
+      try {
+        this.loading = true
+        this.orgUsers = await OrganizationService.getUserPermissionsFacilities(this.userInfo.organizationId)
+      } catch (error) {
+        console.error('Error getting organization users:', error)
+      } finally {
+        this.loading = false
+      }
+    },
+
+    /** 
+     * TODO: Payment type is displayed in UI but is not yet integrated into this method as required CRM data is not yet in place.
+     * This is a know issue and postponed to a future sprint.
+     */
+    async loadFundingAgreements() {
+      try {
+        this.loading = true
+        this.fundingAgreements = []
+        const facilities = this.selectedFacility ? [this.selectedFacility] : this.userInfo.facilities
+        for (let facility of facilities) {
+          try {
+            let payload
+            if (this.selectedDateFilterType === 'Custom') {
+              payload = await FundingAgreementService.getFAByFacilityIdAndStartFromEndDates(facility.facilityId, this.startDateFrom, this.startDateTo)
+            } else {
+              payload = await FundingAgreementService.getFAByFacilityIdAndStartDateThreshold(facility.facilityId, this.startDateThreshold)
+            }
+            if (!payload) continue
+            await this.enrichFundingAgreementData(payload)
+            this.fundingAgreements.push(payload)
+          } catch (error) {
+            console.error(`Error getting funding agreements for facility ID ${facility.facilityId}:`, error)
+          }
+        }
+      } catch (error) {
+        console.error(`Error getting funding agreements:`, error)
+      } finally {
+        this.loading = false
+      }
+    },
+
+    async search() {
+      this.$refs.faForm?.validate()
+      if (!this.isFormComplete) {
+        return
+      }
+      await this.loadFundingAgreements()
+    },
+
+    async enrichFundingAgreementData(result) {
+      result.fundingAgreementType = this.PAYMENT_FILTER_TYPE_VALUES.BASE_FUNDING
+      result.facility = this.facilityName(result.facilityId)
+      result.status = this.statusName(result.statusCode)
+      result.expenseAuthorityName = await this.getExpenseAuthorityName(result.applicationId)
```

Comment:
since the FA entity is linked to Application entity, I think we can expand the backend query to include the expenseAuthority instead of having a separate API call to get application. (e.g.: ofm_fundings?$select=ofm_fundingid,ofm_funding_number,ofm_declaration,ofm_start_date,ofm_end_date,_ofm_application_value,_ofm_facility_value,statuscode,statecode&$expand=ofm_application($select=_ofm_expense_authority_value)&$filter=(_ofm_facility_value eq c9ecef6c-437d-ee11-8179-000d3a09d132 and statecode eq 0)

Reply:
Correct, the backend calls can be reduced with expands in the OData query, though this will require CRM developer support to add. This is what Wes's team comment that we'll likely need to optimize this in the future. Suggest we leave for now as Wes's suggestion to address later.

Reply:
I'm gonna add this suggestion to the Funding agreement tech spike.

Reply:
when I try this query, I can see the expenseAuthorityId in the response. I think it is sufficient?
ofm_fundings?$select=ofm_fundingid,ofm_funding_number,ofm_declaration,ofm_start_date,ofm_end_date,_ofm_application_value,_ofm_facility_value,statuscode,statecode&$expand=ofm_application($select=_ofm_expense_authority_value)&$filter=(_ofm_facility_value eq c9ecef6c-437d-ee11-8179-000d3a09d132 and statecode eq 0)

Reply:
![image](https://github.com/bcgov/ecc-ofm/assets/119005543/d3de5719-e010-46b3-943c-6a20ab4d59a1)